Bugzilla.pm

```diff
@@ -787,15 +787,36 @@ sub check_rate_limit {
       if ($filter && $filter->test($ip)) {
         $action = 'ignore';
       }
-      my $limit = join("/", @$limit);
+      my $full_limit = join("/", @$limit);
       Bugzilla->audit(
-        "[rate_limit] action=$action, ip=$ip, limit=$limit, name=$name");
+        "[rate_limit] action=$action, ip=$ip, limit=$full_limit, name=$name");
       if ($action eq 'block') {
         request_cache->{mojo_controller}->block_ip($ip);
         $throw_error->();
       }
     }
   }
+
+  # Send information about this event to the iprepd API if active
```

Comment:
As discussed on slack informing iprepd about failed authentication attempts should not be part of the rate limiting logic.

Decouple this completely from rate limiting; add a method to report the violation to iprepd, and call that directly from the points in the code where authentication failed.